codec-http2/src/main/java/io/netty5/handler/codec/http2/DefaultHttp2StreamChannel.java

```diff
@@ -131,8 +149,25 @@ private enum ReadStatus {
         REQUESTED
     }
 
+    private static final AtomicIntegerFieldUpdater<DefaultHttp2StreamChannel> AUTOREAD_UPDATER =
+            AtomicIntegerFieldUpdater.newUpdater(DefaultHttp2StreamChannel.class, "autoRead");
+    private static final AtomicReferenceFieldUpdater<DefaultHttp2StreamChannel, WriteBufferWaterMark>
+            WATERMARK_UPDATER = AtomicReferenceFieldUpdater.newUpdater(
+                    DefaultHttp2StreamChannel.class, WriteBufferWaterMark.class, "writeBufferWaterMark");
+
+    private volatile BufferAllocator bufferAllocator = DefaultBufferAllocators.preferredAllocator();
+    private volatile RecvBufferAllocator rcvBufAllocator = new AdaptiveRecvBufferAllocator();
+
+    private volatile int connectTimeoutMillis = 30000;
+    private volatile int writeSpinCount = 16;
+    private volatile int maxMessagesPerWrite = Integer.MAX_VALUE;
```

Comment:
It may be worth it to have another abstract class below `DefaultAttributeMap`, so we can share more code between here and `AbstractChannel`.